### PR TITLE
[TASK] Reduce Node complexity and method visibility

### PR DESCRIPTION
**Pull request by kitsunet**
This is mostly a cleanup that tries to split the ``setPath``
method into smaller parts. On the way the visibility of some
methods was revised.

Releases: master

### CODE REVIEW

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php [hunk 1]
```diff
@@ -145,70 +146,137 @@ public function setName($newName) {
...
-
-			/** @var NodeData $nodeData */
-			foreach ($nodeDataVariantsAndChildren as $nodeData) {
-				$nodeVariant = $this->createNodeForVariant($nodeData);
-				if ($nodeVariant !== NULL) {
-					$relativePathSegment = NodePaths::getRelativePathBetween($originalPath, $nodeVariant->getPath());
-					$newNodeVariantPath = NodePaths::addNodePathSegment($path, $relativePathSegment);
-					$possibleShadowedNodeData = $nodeData->move($newNodeVariantPath, $this->context->getWorkspace());
-					$nodeVariant->setNodeData($possibleShadowedNodeData);
-					$changedNodePathsCollection[] = array($nodeVariant, $originalPath, $nodeVariant->getNodeData()->getPath(), !$checkForExistence);
```
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php]: Actually !$checkForExistence was wrong here, as  that meant that setting path on an aggregate would negate it for all changes and so no nodePathChanged signal with this flag would ever be emitted.

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php [hunk 2]
```diff
@@ -115,6 +115,10 @@ public function getContextPath() {
 	 * @api
 	 */
 	public function setName($newName) {
+		if ($this->getName() === $newName) {
+			return;
+		}
+
```
Comment (bwaidelich) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php]: Is that related to this change really?

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php [hunk 3]
```diff
@@ -145,70 +146,135 @@ public function setName($newName) {
 	 *
 	 * @param string $path
 	 * @param boolean $checkForExistence Checks for existence at target path, internally used for recursions and shadow nodes.
+	 * @return void
 	 * @throws NodeException
-	 * @internal This method is purely internal and Node objects can call this on other Node objects
 	 */
-	public function setPath($path, $checkForExistence = TRUE) {
+	protected function setPath($path, $checkForExistence = TRUE) {
```
Comment (bwaidelich) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php]: Wasn't that method used from the "outside" at all? Is it safe to make it protected?

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php (hunk 1 above)
Comment (bwaidelich) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php]: I don't get what you mean by that exactly, but I would suggest to separate refactoring from modified behavior to make this one easier to review & test

**Comment by bwaidelich** [comment]
I left some comments, but couldn't really add valuable input
This hasn't been updated for a long time - any news/impediments?

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php (hunk 3 above)
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php]: That's the trick, it is absolutely valid to call a protected method on a instance of the same class other than $this. And it is not called from anywhere else outside of Node.

Comment (bwaidelich) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php]: mind.blown. I bloody didn't know that! o.O
It's even allowed for private methods.. Good to know ;)

File: TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php (hunk 2 above)
Comment (kitsunet) [inline TYPO3.TYPO3CR/Classes/TYPO3/TYPO3CR/Domain/Model/Node.php]: Yes, it was moved further up, see full diff. It makes sense to check that first before anything else.

**Comment by kitsunet** [comment]
rebased

**Comment by kitsunet** [comment]
Travis seems to be broken with this... Will open a new PR, sry.
